froguro/CSE110-TritonFitness
# Setup Backend for Emotion Tracker

# Setup Backend for Emotion Tracker 

## Changes Made
- Set up TypeScript Express server running on port 3001
- Implemented SQLite database with initial emotions table
- Created `/api/emotions` endpoint to fetch all available emotions

## Frontend Integration 
The frontend emotion tracker UI can now:
- Fetch available emotions using the `/api/emotions` endpoint
- Display emotion options to users
- Access predefined emotions like 'happy', 'sad', 'anxious', etc.

## Next Steps
- Create users table for authentication
- Implement emotion_logs table to track user emotions over time
- Add endpoints for logging and retrieving user emotions

## Testing
To test the backend:
1. Run `npm install` to install dependencies
2. Start the server with `npm start`
3. Access `http://localhost:3001/api/emotions` to see available emotions
